redis/src/main/scala/akka/streams/alpakka/redis/scaladsl/RedisSource.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.streams.alpakka.redis.scaladsl
+
+import java.util.function.Predicate
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import akka.streams.alpakka.redis.{RedisHSet, RedisKeyValue, RedisPubSub}
+import io.lettuce.core.KeyValue
+import io.lettuce.core.api.StatefulRedisConnection
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection
+import io.lettuce.core.pubsub.api.reactive.ChannelMessage
+import reactor.core.publisher.Flux
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable.Seq
+
+object RedisSource {
+
+  def subscribe[K, V](topics: Seq[K],
+                      connection: StatefulRedisPubSubConnection[K, V]): Source[RedisPubSub[K, V], NotUsed] = {
+    connection.reactive().subscribe(topics: _*).subscribe()
+    Source
+      .fromPublisher(
+        connection
+          .reactive()
+          .observeChannels()
+          .filter(new Predicate[ChannelMessage[K, V]] {
+            override def test(t: ChannelMessage[K, V]): Boolean = topics.contains(t.getChannel)
+          })
+      )
+      .map(f => RedisPubSub(f.getChannel, f.getMessage))
+  }
+
+  def get[K, V](key: K, connection: StatefulRedisConnection[K, V]): Source[RedisKeyValue[K, V], NotUsed] =
+    Source.fromPublisher(connection.reactive().get(key)).map(f => RedisKeyValue(key, f))
```

Comment:
This could be implemented using `RedisFlow` in the same way as all `RedisSink` operations are using `RedisFlow`.